src/primitives/frustumsurface.jl

```diff
@@ -31,6 +31,35 @@ top(f::FrustumSurface) = f.top
 
 height(f::FrustumSurface) = norm(center(bottom(f)) - center(top(f)))
 
+paramdim(::Type{<:FrustumSurface}) = 2
+
+axis(f::FrustumSurface) = Line(center(bottom(f)), center(top(f)))
+
+function (f::FrustumSurface{T})(φ, z) where {T}
+  if (φ < 0 || φ > 1) || (z < 0 || z > 1)
+    throw(DomainError((φ, z), "f(φ, z) is not defined for φ, z outside [0, 1]²."))
+  end
+  rb = radius(bottom(f))
+  rt = radius(top(f))
+  a = axis(f)
+  d = a(1) - a(0)
+  l = norm(d)
+
+  # rotation to align z axis with cylinder axis
+  Q = rotation_between(d, Vec{3,T}(0, 0, 1))
+
+  # scale coordinates
+  φₛ = 2T(π) * φ
+  zₛ = z * l
+
+  x_local = cos(φₛ) * (rb * (l - zₛ) + rt * zₛ) / l
+  y_local = sin(φₛ) * (rb * (l - zₛ) + rt * zₛ) / l
+  z_local = zₛ
+  p_local = Vec{3,T}(x_local, y_local, z_local)
+
+  center(bottom(f)) + Q' * p_local
```

Comment:
please avoid underscores in variable names

Reply:
I've done a force psuh with the fix.